### PR TITLE
Add type annotations to `torch_frame.utils.*`

### PR DESCRIPTION
**Pull request by akihironitta**
🎅🎁

### CODE REVIEW

File: torch_frame/utils/concat.py [hunk 1]
```diff
@@ -11,11 +12,27 @@
 from torch_frame.data.tensor_frame import TensorFrame
 from torch_frame.typing import TensorData
 
+T = TypeVar(
+    "T",
```
Comment (zechengz) [inline torch_frame/utils/concat.py:16]: What's the meaning of T here?

**Review by zechengz — APPROVED** [review]

File: torch_frame/utils/concat.py (hunk 1 above)
Comment (akihironitta) [reply torch_frame/utils/concat.py:16]: I think `T` is a common name for a type variable although I'm open to renaming it :)

File: torch_frame/utils/concat.py [hunk 2]
```diff
@@ -46,7 +64,7 @@ def _cat_tensor_data(td_list: list[TensorData], dim: int) -> TensorData:
     if len(td_list) == 0:
         raise ValueError("Cannot concatenate an empty list.")
     elif len(td_list) == 1:
-        return td_list[0]
+        return td_list[0]  # TensorData
```
Comment (akihironitta) [inline torch_frame/utils/concat.py:67]:
```suggestion
        return td_list[0]
```

File: torch_frame/utils/concat.py (hunk 1 above)
Comment (zechengz) [reply torch_frame/utils/concat.py:17]: Oh, sorry I mean the `"T"` in `TypeVar`. Didn't know that we should have `"T"` in `TypeVar` to match with the `T`

**Pull request merged**